test/test_transforms_v2_refactored.py

```diff
@@ -1432,10 +1376,269 @@ def test_bounding_box_correctness(self, format, fn):
+    @pytest.mark.parametrize("dtype", [torch.float32, torch.uint8])
+    @pytest.mark.parametrize("device", cpu_and_cuda())
+    def test_kernel_bounding_box(self, param, value, format, dtype, device):
+        kwargs = {param: value}
+        if param != "angle":
+            kwargs["angle"] = -10.9
+
+        bounding_box = make_input(datapoints.BoundingBox, dtype=dtype, device=device, format=format)
+
+        check_kernel(
+            F.rotate_bounding_box,
+            bounding_box,
+            format=format,
+            spatial_size=bounding_box.spatial_size,
+            **kwargs,
+        )
+
+    @pytest.mark.parametrize("mask_type", ["segmentation", "detection"])
+    def test_kernel_mask(self, mask_type):
+        check_kernel(F.rotate_mask, make_input(datapoints.Mask, mask_type=mask_type), **self._MINIMAL_AFFINE_KWARGS)
+
+    def test_kernel_video(self):
+        check_kernel(F.rotate_video, make_input(datapoints.Video), **self._MINIMAL_AFFINE_KWARGS)
+
+    @pytest.mark.parametrize(
+        ("input_type", "kernel"),
+        [
+            (torch.Tensor, F.rotate_image_tensor),
+            (PIL.Image.Image, F.rotate_image_pil),
+            (datapoints.Image, F.rotate_image_tensor),
+            (datapoints.BoundingBox, F.rotate_bounding_box),
+            (datapoints.Mask, F.rotate_mask),
+            (datapoints.Video, F.rotate_video),
+        ],
+    )
+    def test_dispatcher(self, kernel, input_type):
+        check_dispatcher(F.rotate, kernel, make_input(input_type), **self._MINIMAL_AFFINE_KWARGS)
+
+    @pytest.mark.parametrize(
+        ("input_type", "kernel"),
+        [
+            (torch.Tensor, F.rotate_image_tensor),
+            (PIL.Image.Image, F.rotate_image_pil),
+            (datapoints.Image, F.rotate_image_tensor),
+            (datapoints.BoundingBox, F.rotate_bounding_box),
+            (datapoints.Mask, F.rotate_mask),
+            (datapoints.Video, F.rotate_video),
+        ],
+    )
+    def test_dispatcher_signature(self, kernel, input_type):
+        check_dispatcher_signatures_match(F.rotate, kernel=kernel, input_type=input_type)
+
+    @pytest.mark.parametrize(
+        "input_type",
+        [torch.Tensor, PIL.Image.Image, datapoints.Image, datapoints.BoundingBox, datapoints.Mask, datapoints.Video],
+    )
+    @pytest.mark.parametrize("device", cpu_and_cuda())
+    def test_transform(self, input_type, device):
+        input = make_input(input_type, device=device)
+
+        check_transform(transforms.RandomRotation, input, **self._CORRECTNESS_TRANSFORM_AFFINE_RANGES)
+
+    @pytest.mark.parametrize("angle", _CORRECTNESS_AFFINE_KWARGS["angle"])
+    @pytest.mark.parametrize("center", _CORRECTNESS_AFFINE_KWARGS["center"])
+    @pytest.mark.parametrize(
+        "interpolation", [transforms.InterpolationMode.NEAREST, transforms.InterpolationMode.BILINEAR]
+    )
+    # TODO: investigate why expand=True leads to different shapes between PIL and tensor
```

Comment:
Seems important. Will have a look soon.

Reply:
PIL is doing the computation internally, but according to our function, we should do exactly the same:

https://github.com/pytorch/vision/blob/71968bc4afb8892284844a7c4cbd772696f42a88/torchvision/transforms/v2/functional/_geometry.py#L443-L445

However, this function was "optimized" to a point where we might as well slightly deviate from the PIL implementation. I'll have a look.

Reply:
#7714

Reply:
Let's wait for #7715 so we don't have to skip here.